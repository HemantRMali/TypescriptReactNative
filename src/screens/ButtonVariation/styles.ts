import {StyleSheet} from 'react-native';
import {colors} from '../../assets/colors';
export default StyleSheet.create({
  container: {flex: 1, backgroundColor: 'black', opacity: 0.9},
  bottomView: {
    width: '100%',
    position: 'absolute',
    bottom: 50,
  },

  textStyle: {
    color: '#fff',
    fontSize: 22,
  },
  containerStyles: {borderRadius: 10, margin: 20},
  thumbIconStyles: {borderRadius: 10},
  railStyles: {backgroundColor: colors.black, borderColor: colors.black},
  title: {color: colors.yellow, alignSelf: 'center'},
  screenTItle: {fontSize: 25, alignSelf: 'flex-end', color: colors.white},
});
