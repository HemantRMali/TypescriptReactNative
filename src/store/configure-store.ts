import {createStore, applyMiddleware} from 'redux';
import {createLogger} from 'redux-logger';
// Imports: Redux Root Reducer
import appReducer from '../reducers';

// Redux: Store
const store = createStore(appReducer, applyMiddleware(createLogger()));

// Exports
export default store;
