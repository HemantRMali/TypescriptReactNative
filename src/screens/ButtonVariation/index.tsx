import * as React from 'react';
import {Text, View, Image} from 'react-native';
import DefaultButton from '../../components/DefaultButton/DefaultButton';
import SwipeButton from 'rn-swipe-button';
import ScreenTitle from '../../components/ScreenTitle/ScreenTitle';
import {useSelector} from 'react-redux';
import {RootState} from '../../reducers/reducers';
import styles from './styles';
import {Strings} from '../../constants/constants';
import colors from '../../assets/colors/colors';

import diamondIcon from '../../assets/images/diamond.png';
import {RootStackParamList} from '../../../App';
import {StackNavigationProp} from '@react-navigation/stack';

type ButtonVariationScreenNavigationProp = StackNavigationProp<
  RootStackParamList,
  'ButtonVariations'
>;

interface ButtonVariationProps {
  userName?: string;
  navigation?: ButtonVariationScreenNavigationProp;
}

const ButtonVariation: React.FC<ButtonVariationProps> = ({navigation}) => {
  const user = useSelector((state: RootState) => state.UserReducer.user);
  const thumbIcon = () => <Image resizeMode="center" source={diamondIcon} />;

  const navigateToDetails = () => {
    navigation?.navigate('Details');
  };
  return (
    <View style={styles.container}>
      <ScreenTitle style={styles.screenTItle} title={user.name} />

      <View style={styles.bottomView}>
        <Text style={styles.title}>4 variations of a button</Text>
        <DefaultButton
          title={Strings.pressMe}
          onPress={navigateToDetails}
          style={styles.secondary}
          textStyle={styles.textStyle}
        />
        <DefaultButton
          title={Strings.pressMe}
          onPress={navigateToDetails}
          style={styles.third}
          textStyle={styles.textStyle}
        />
        <DefaultButton
          title={Strings.pressMe}
          onPress={navigateToDetails}
          style={styles.primary}
        />
        <SwipeButton
          containerStyles={styles.containerStyles}
          thumbIconBackgroundColor={colors.steelblue}
          thumbIconStyles={styles.thumbIconStyles}
          thumbIconComponent={thumbIcon}
          title={Strings.slideToContinue}
          titleColor={colors.white}
          railStyles={styles.railStyles}
          railBackgroundColor={colors.black}
          onSwipeSuccess={() => navigateToDetails()}
          shouldResetAfterSuccess={true}
        />
      </View>
    </View>
  );
};

export default ButtonVariation;
