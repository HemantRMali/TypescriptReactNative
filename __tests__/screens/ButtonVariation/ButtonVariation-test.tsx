// __tests__/ButtonVariation-test.js
import React from 'react';
import ButtonVariation from '../../../src/screens/ButtonVariation';
import {render, fireEvent} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import store from '../../../src/store/configure-store';
import SwipeButton from 'rn-swipe-button';

describe('ButtonVariation', () => {
  it('renders properly', () => {
    const {toJSON, UNSAFE_getByType} = render(
      <Provider store={store}>
        <ButtonVariation />
      </Provider>,
    );

    const sliderButton = UNSAFE_getByType(SwipeButton);
    fireEvent(sliderButton, 'onSwipeSuccess');
    expect(toJSON()).toMatchSnapshot();
  });
});
