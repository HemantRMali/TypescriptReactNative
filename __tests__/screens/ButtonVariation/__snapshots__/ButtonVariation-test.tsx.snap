// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonVariation renders properly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
      "opacity": 0.9,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#595856",
          "fontSize": 40,
          "margin": 20,
        },
        Object {
          "alignSelf": "flex-end",
          "color": "white",
          "fontSize": 25,
        },
      ]
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "bottom": 50,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "yellow",
        }
      }
    >
      4 variations of a button
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "justifyContent": "center",
              "margin": 20,
              "padding": 15,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "color": "steelblue",
                "fontSize": 22,
              },
            ]
          }
        >
          Press me
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "justifyContent": "center",
              "margin": 20,
              "padding": 15,
            },
            Object {
              "backgroundColor": "lightgray",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "color": "steelblue",
                "fontSize": 22,
              },
            ]
          }
        >
          Press me
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "justifyContent": "center",
              "margin": 20,
              "padding": 15,
            },
            Object {
              "backgroundColor": "steelblue",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
            ]
          }
        >
          Press me
        </Text>
      </View>
    </View>
    <View />
  </View>
</View>
`;
